patches/chromium/replace_clearfilterdata_with_invalidatefilterdata.patch

```diff
@@ -0,0 +1,133 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Raymond Zhao <raymondzhao@microsoft.com>
+Date: Wed, 10 Feb 2021 12:03:34 -0800
+Subject: Replace ClearFilterData with InvalidateFilterData
+
+This is a hand-patch of https://chromium-review.googlesource.com/c/chromium/src/+/2566992.
+
+Firstly, the InvalidateFilterData function was copied over. Then, the ClearFilterData function was deleted, and any calls to it were replaced with InvalidateFilterData. Redundant calls were removed.
+
+The InvalidateFilterData function in this patch contains a fix for SVG content sometimes failing to appear after modification https://bugs.chromium.org/p/chromium/issues/detail?id=1154050.
+
+diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.cc b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.cc
+index c7093ea1724dc99a26c3eef033d7ef13c2888f54..8bc6ab492bba91b5c155982e1b145aa68f43dd97 100644
+--- a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.cc
++++ b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.cc
+@@ -227,16 +227,12 @@ static inline void RemoveFromCacheAndInvalidateDependencies(
+     if (resources->HasClipOrMaskOrFilter()) {
+       InvalidationModeMask invalidation_mask =
+           SVGResourceClient::kBoundariesInvalidation;
+-      bool filter_data_invalidated = false;
+       if (resources->Filter()) {
+-        filter_data_invalidated = client->ClearFilterData();
+-        invalidation_mask |=
+-            filter_data_invalidated ? SVGResourceClient::kPaintInvalidation : 0;
++        client->InvalidateFilterData();
++        invalidation_mask |= SVGResourceClient::kPaintInvalidation;
```

Comment:
`InvalidateFilterData` calls into `layout_object->SetNeedsPaintPropertyUpdate` and I don't see `SVGResourceClient::kPaintInvalidation` mask set in https://chromium-review.googlesource.com/c/chromium/src/+/2454053 , is the mask really needed in combination with `InvalidateFilterData`  ?

Reply:
I'll leave the `MarkClientForInvalidation` code along with the masks/flags for now, as adding in the https://chromium-review.googlesource.com/c/chromium/src/+/2454053 patch requires a lot of changes, for example adding in an entire `SVGResourceInvalidator`.